.github/7-sprint-mission-fe/week1/loginpage/login.js

```diff
@@ -1,26 +1,153 @@
+// 눈 아이콘
+document.querySelectorAll('.eyes').forEach(eye => {
+  eye.addEventListener('click', () => {
+    const inputWrap = eye.closest('.input_wrap');
+    const passwordInput = inputWrap.querySelector('input');
+    const eyeIcon = eye.querySelector('i');
 
-eyes.forEach((eye) => {
+    inputWrap.classList.toggle('active');
+    const isActive = inputWrap.classList.contains('active');
+    eyeIcon.className = isActive ? 'fas fa-eye' : 'fas fa-eye-slash';
+    passwordInput.type = isActive ? 'text' : 'password';
+  });
+});
 
-  eye.addEventListener('click', function () {
 
-    const inputWrap = eye.closest('.input_wrap'); 
 
-    const passwordInput = inputWrap.querySelector('input');
-    const eyeIcon = eye.querySelector('i');
+//email
+function isValidEmail(email) {
+  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+  return regex.test(email);
+}
 
-    inputWrap.classList.toggle('active');
+
+function validateEmail() {
+  const email = document.getElementById('email');
+  const errorDiv = document.getElementById('email-error');
+  const value = email.value.trim();
+  if (!value) {
+    email.classList.add('error');
+    errorDiv.textContent = '이메일을 입력해주세요.';
+    return false;
+  } else if (!isValidEmail(value)) {
+    email.classList.add('error');
+    errorDiv.textContent = '잘못된 이메일 형식입니다.';
+    return false;
+  } else {
+    email.classList.remove('error');
+    errorDiv.textContent = '';
+    return true;
+  }
+}
+
+document.getElementById('email').addEventListener('blur', () => {
+  validateEmail();
+  checkFormValidity();
+});
 
 
-    if (inputWrap.classList.contains('active')) { 
-      eyeIcon.className = 'fas fa-eye';
-      passwordInput.type = 'text';
-    } else {
-      eyeIcon.className = 'fas fa-eye-slash';
-      passwordInput.type = 'password';
-    }
-  });
 
+// password
+function validatePassword() {
+  const password = document.getElementById('password');
+  const errorDiv = document.getElementById('password-error');
+  const value = password.value;
+
+  if (!value) {
+    password.classList.add('error');
+    errorDiv.textContent = '비밀번호를 입력해주세요.';
+    return false;
+  } else if (value.length < 8) {
+    password.classList.add('error');
+    errorDiv.textContent = '비밀번호를 8자 이상 입력해주세요.';
+    return false;
+  } else {
+    password.classList.remove('error');
+    errorDiv.textContent = '';
+    return true;
+  }
+}
+document.getElementById('password').addEventListener('blur', () => {
+  validatePassword();
+  checkFormValidity();
 });
 
+
+
+
+// 포커스 시 에러 제거
+function removeErrorOnFocus(id, errorId) {
+  const input = document.getElementById(id);
+  const errorDiv = document.getElementById(errorId);
+
+  input.addEventListener('focus', () => {
+    input.classList.remove('error');
+    errorDiv.textContent = '';
+  });
+}
+// focus 이벤트 등록 (빨간 테두리 제거)
+removeErrorOnFocus('email', 'email-error');
+removeErrorOnFocus('password', 'password-error');
+
+
+
+
+
+
+
+const USER_DATA = [
```

Comment:
```suggestion
const users = [
```

뭐 users로 하던 USERS로 하던 어차피 DATA는 필요없는 postfix이기 때문에 붙히지 않아도 될 것 같습니다.